cmd/openvdc-executor/sshd.go

```diff
@@ -22,9 +23,52 @@ type SSHServer struct {
 	ctx      context.Context
 }
 
+func getInstResource(ctx context.Context, userID string) (model.InstanceResource, error) {
+	inst, err := model.Instances(ctx).FindByID(userID)
+	if err != nil {
+		log.WithError(err).Errorf("Unknown instance: %s", userID)
+		// conn.Close()
+		return nil, err
+	}
+	instResource := inst.ResourceTemplate().(model.InstanceResource)
+	return instResource, nil
+}
+
 func NewSSHServer(provider hypervisor.HypervisorProvider, ctx context.Context) *SSHServer {
 	config := &ssh.ServerConfig{
-		NoClientAuth: true,
+		PasswordCallback: func(conn ssh.ConnMetadata, pass []byte) (*ssh.Permissions, error) {
```

Comment:
Why is the password authentication enabled? Just ``NONE`` and ``PUB_KEY`` are listed in ``model.proto``.

Reply:
Library of SSH can't call PublickeyCallBack if does not  designate public key. Therefore if client don't designate public key then server authenticate client by using PasswordCallback without password.